### DISCUSSION TITLE
Questions on ce_MonoSynConvClick

### DISCUSSION THREAD

**Discussion by oaaij-gnahz** [petersenpeter/CellExplorer#68] [General]
Hi, I was trying to use `ce_MonoSynConvClick` to calculate putative connectivity between units. While doing that I found the following problem: say I have unit A and B, the function would sometimes return A->B as both excitatory and inhibitory connectivity. I looked at the code, and it turns out from [line 217 to 231](https://github.com/petersenpeter/CellExplorer/blob/master/calc_CellMetrics/ce_MonoSynConvClick.m#L217) that it was looking for excitatory connections using the maximum value of the cross-correlogram in the range indicated by `postbins`. If that maximum value satisfies 2 conditions then it is determined to be an excitatory connection.
Similarly from [line 269 to 282](https://github.com/petersenpeter/CellExplorer/blob/master/calc_CellMetrics/ce_MonoSynConvClick.m#L269), we can see it was looking for inhibitory connections merely by looking at the minimum value of cross-correlogram in `postbins`.
With that said, if a CCH has really high fluctuation in range `postbins` (it seems like 8 bins by default if I understood correctly), then the same pair of units A->B (A as the source) can be determined as both excitatory and inhibitory, which is undesirable. What are your opinions/suggestions here?

Thank you so much!

**Comment by petersenpeter** [comment]
Hi

Would you like to have a post-decision built-in to handle these "extreme" cases, where connections are determined to be both excitatory and inhibitory? I believe this is mostly an issue with short recordings and with units with few spikes. That said, the algorithm does require manual curation, to remove detected connections that are significant but merely due to synchronization, or noise in the data. 

The detection method definitely has room for improvement (due to the above-mentioned shortcomings), but what would the post-decision be in your opinion?

**Reply by oaaij-gnahz** [reply]
Thanks!

I agree this is probably an issue with units with few spikes in this case. For my situation I would probably discard those connections. It shouldn't hurt since we are only looking for general patterns in a large population of units. For more general scenario, I do agree that manual curation is a safe approach even though it might take a lot of labor. 

I have a tentative idea (not implemented yet): besides checking the CCH criterion, we also look at the correlation between the pair's "continuous" firing rate (binned spikes) time-series (e.g. a threshold on the absolute value of Pearson's R or cross-correlation). We call a connection only if both CCH and firing rate criteria are satisfied. The reasoning is that we assume if 2 units are connected then their firing patterns should show some correlation/synchronization (which I haven't tested). The firing rates criterion may serve as a more robust but less accurate approach compared to CCH. So they kind of complement each other. Do you think it is a reasonable assumption/what would be your comments?

Thank you

**Comment by petersenpeter** [comment]
I don't know if a correlated firing rate is a good measure, but I know of others that use the waveform width to filter the detected connections (in the hippocampus, the excitatory connections are from the "wide waveform" pyramidal cells to the "narrow waveform" interneurons).
